Sources/JavaScriptKit/BasicObjects/JSArray.swift

```diff
@@ -44,6 +48,35 @@ extension JSArray: RandomAccessCollection {
     public var startIndex: Int { 0 }
 
     public var endIndex: Int { ref.length.number.map(Int.init) ?? 0 }
+
+    /// The number of elements in that array including empty hole.
+    /// Note that `length` respects JavaScript's `Array.prototype.length`
+    ///
+    /// e.g.
+    /// ```javascript
+    /// const array = [1, , 3];
+    /// ```
+    /// ```swift
+    /// let array: JSArray = ...
+    /// array.length // 3
+    /// array.count  // 2
+    /// ```
+    public var length: Int {
+        return Int(ref.length.number!)
+    }
+
+    /// The number of elements in that array **not** including empty hole.
+    /// Note that `count` syncs with the number that `Iterator` can iterate.
+    /// See also: `JSArray.length`
+    public var count: Int {
+        return getObjectValuesLength(ref)
+    }
+}
+
+private func getObjectValuesLength(_ object: JSObject) -> Int {
+    let objectClass = JSObject.global.Object.function!
+    let values = objectClass.values!(object).object!
```

Comment:
This won't work in another edge case — when there are additional properties added to the array (like doing `array.foo = 'bar'`).

According to [this post](https://2ality.com/2015/09/holes-arrays-es6.html), running `array.filter(() => true)` removes the holes so that could be used to get a better count.